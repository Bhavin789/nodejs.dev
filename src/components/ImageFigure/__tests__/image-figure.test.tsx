/* eslint-disable @typescript-eslint/explicit-function-return-type */
import React from 'react';
import renderer from 'react-test-renderer';
import ImageFigure from '..';

describe('ImageFigure component', () => {
  it('renders correctly with footer', () => {
    const tree = renderer
      .create(
        <ImageFigure
          caption="mock-caption"
          src="mock-src"
          target="_self"
          alt="mock-alternate"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
