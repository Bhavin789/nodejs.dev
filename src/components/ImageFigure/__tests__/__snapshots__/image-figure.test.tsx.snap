// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageFigure component renders correctly with footer 1`] = `
<figure>
  <a
    href="mock-src"
    rel="noopener noreferrer"
    target="_self"
  >
    <img
      alt="mock-alternate"
      className="figure__image"
      src="mock-src"
    />
  </a>
  <figcaption>
    mock-caption
  </figcaption>
</figure>
`;
